src/realm/array_basic_tpl.hpp

```diff
@@ -163,6 +163,19 @@ inline void BasicArray<T>::set(size_t ndx, T value)
 }
 
 template <class T>
+inline void BasicArray<T>::initialize(size_t ndx, T value)
+{
+    REALM_ASSERT_3(ndx, <, m_size);
+
+    // Check if we need to copy before modifying
+    copy_on_write(); // Throws
+
+    // Set the value
+    T* data = reinterpret_cast<T*>(m_data) + ndx;
+    *data = value;
+}
+
```

Comment:
If this function is only meant to be used in `create_array()`, we already know that `ndx < m_size`. Furthermore, is it necessary to call `copy_on_write`, when the array has just been created? It seems to me that inlining this code might be a better option all together?